app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -237,8 +237,23 @@ function FormSteps({
       const { element, isRequired } = fields.current[key];
       const isActive = !!element;
 
-      if (isActive && isRequired && !values[key]) {
-        result = result.concat({ field: key, error: new RequiredValueMissingError() });
+      let error: Error | undefined;
+      if (isActive) {
+        const validatedField = element.closest('lg-validated-field');
+        if (validatedField) {
+          validatedField.validate();
+        }
+
+        if (element.validationMessage) {
+          error = new Error(element.validationMessage);
+          element.reportValidity();
+        } else if (isRequired && !values[key]) {
```

Comment:
Future refactoring could see about eliminating this custom idea of `isRequired` and just use `element.validityState.valueMissing` in combination with `<input required>`.